src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -29,17 +29,76 @@
 
 #include <drivers/hwinfo.h>
 
+#ifdef CONFIG_MCUBOOT_IMG_MANAGER
+#include <dfu/mcuboot.h>
+#endif
+
 #include <malloc.h>
 
 namespace chip {
 namespace DeviceLayer {
 
+namespace {
+
+uint8_t DetermineBootReason()
```

Comment:
```suggestion
BootReasonType DetermineBootReason()
```

Reply:
I can't do it because `DiagnosticDataProvider::GetBootReason` takes `uint8_t&`. It probably can be cleaned up now that we have `ClusterEnums.h`. I'll try to address that in another PR (not Zephyr-specific).

Reply:
Posted: #16556